src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/HttpCacheProxyFactory.java

```diff
@@ -275,92 +278,185 @@ private MessageConsumer newNativeInitialStream(
+                                                  requestHash,
+                                                  requestURL,
+                                                  acceptReply,
+                                                  acceptRouteId,
+                                                  acceptReplyId,
+                                                  acceptInitialId,
+                                                  connectInitial,
+                                                  connectReply,
+                                                  connectInitialId,
+                                                  connectReplyId,
+                                                  connectRouteId);
+        newStream = nonCacheableRequest::onRequestMessage;
+        correlations.put(connectReplyId, nonCacheableRequest::newResponse);
+        router.setThrottle(acceptReplyId, nonCacheableRequest::onResponseMessage);
+        return newStream;
+    }
+
+    private MessageConsumer createCacheableRequestStream(
+        ArrayFW<HttpHeaderFW> requestHeaders,
+        MessageConsumer acceptReply,
+        long connectRouteId,
+        long acceptInitialId,
+        long acceptRouteId,
+        String requestURL,
+        int requestHash,
+        long connectInitialId,
+        MessageConsumer connectInitial,
+        long connectReplyId,
+        long acceptReplyId)
+    {
+        counters.requestsCacheable.getAsLong();
+        MessageConsumer newStream;
+        HttpProxyCacheableRequestGroup group = requestGroups.computeIfAbsent(requestHash, this::newCacheableRequestGroup);
+
+        HttpHeaderFW authorizationHeader = requestHeaders.matchFirst(h -> AUTHORIZATION.equals(h.name().asString()));
+        if (authorizationHeader != null)
+        {
+            group.setRecentAuthorizationToken(authorizationHeader.value().asString());
+        }
+        final HttpCacheProxyCacheableRequest cacheableRequest =
+            new HttpCacheProxyCacheableRequest(this,
+                                                group,
+                                                requestHash,
+                                                requestURL,
+                                                acceptReply,
+                                                acceptRouteId,
+                                                acceptInitialId,
+                                                acceptReplyId,
+                                                connectInitial,
+                                                connectInitialId,
+                                                connectReplyId,
+                                                connectRouteId);
+        correlations.put(connectReplyId, cacheableRequest::newResponse);
+        newStream = cacheableRequest::onAcceptMessage;
+        router.setThrottle(acceptReplyId, newStream);
+        return newStream;
+    }
+
+    private void handIsOnlyCached(
```

Comment:
This method should probably be renamed for clarity.

Reply:
Done.